packages/backend/src/Backend.hs

```diff
@@ -0,0 +1,96 @@
+module Backend(BackendArgs(..), Target(..), runBackend) where
```

Comment:
Given that we're reorganising all the modules of Happy anyhow, it might be a good idea to give them all hierarchical names while we're at it. How do people feel about just putting them under the `Happy.*` hierarchy like `Happy.Backend` etc?

Reply:
Sounds good, done.
Due to the separation into different cabal packages, it is unfortunately not possible to collect all the source files in one central `src/Happy` directory, which contains `Core`, `Backend`, `Frontend`, `Middleend` and `Test` subfolders. Cabal requires a separation of the packages, leading to a `packages/backend/src/Happy/Backend/...` structure; each package still has its own separate `src` directory.